lib/index.js

```diff
@@ -67,7 +67,7 @@ export async function applySession(
           `next-iron-session: Cookie length is too big ${cookieValue.length}, browsers will refuse it`,
         );
       }
-      const existingSetCookie = [res.getHeader("set-cookie") || []].flat();
+      const existingSetCookie = [res.getHeader('set-cookie') || []].reduce((acc, val) => acc.concat(val), []); //Flat array
```

Comment:
could you create a flattenArray() function instead in the same file (at the bottom) and use it?